zio-http/src/test/scala/zhttp/http/HttpSpec.scala

```diff
@@ -136,11 +136,40 @@ object HttpSpec extends DefaultRunnableSpec with HExitAssertion {
             val actual = (a ++ b).execute(2)
             assert(actual)(isSuccess(equalTo("B")))
           } +
+          test("should resolve third") {
+            val a      = Http.collect[Int] { case 1 => "A" }
+            val b      = Http.collect[Int] { case 2 => "B" }
+            val c      = Http.collect[Int] { case 3 => "C" }
+            val actual = (a ++ b ++ c).execute(3)
+            assert(actual)(isSuccess(equalTo("C")))
+          } +
+          test("should resolve third") {
+            val a      = Http.collectZIO[Int] { case 1 => UIO("A") }
+            val b      = Http.collectZIO[Int] { case 2 => UIO("B") }
+            val c      = Http.collectZIO[Int] { case 3 => UIO("C") }
+            val actual = (a ++ b ++ c).execute(3)
+            assert(actual)(isEffect)
```

Comment:
```suggestion
            assert(actual)(isEffect(equalTo("C")))
```

Reply:
This needs to be resolved. May be we should run the effect and assert on value.

Reply:
This test intends to check if after combining apps, it's reaching the third app that returns an effect. I discussed it with @girdharshubham  and this works

Reply:
I can add this test to assert on effect value:
```scala
          testM("should resolve third") {
            val a      = Http.collect[Int] { case 1 => "A" }
            val b      = Http.collect[Int] { case 2 => "B" }
            val c      = Http.collectZIO[Int] { case 3 => UIO("C") }
            val actual = (a ++ b ++ c).execute(3).toZIO
            assertM(actual)(equalTo("C"))
          } +
```

Reply:
Not needed, marking it as resolved